Mailer/SymfonyAuthCodeMailer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scheb\TwoFactorBundle\Mailer;
+
+use Scheb\TwoFactorBundle\Model\Email\TwoFactorInterface;
+use Symfony\Component\Mailer\Mailer;
+use Symfony\Component\Mime\Email;
+
+class SymfonyAuthCodeMailer implements AuthCodeMailerInterface
+{
+    /**
+     * @var Mailer
+     */
+    private $mailer;
+
+    /**
+     * @var string
+     */
+    private $senderEmail;
+
+    /**
+     * @var string|null
+     */
+    private $senderName;
+
+    public function __construct(Mailer $mailer, string $senderEmail, ?string $senderName)
```

Comment:
You're referencing `Mailer` here, which makes it impossible to mock this dependency since it's a final class. Please use  `MailerInterface` instead and use a mock of `MailerInterface` in the test.